### PR TITLE
Possibility to create custom profile field types via modules

### PR DESCRIPTION
**Pull request by KarlKeu00**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `develop` branch, _not_ the `master` branch if no hotfix
- [x] When resolving a specific issue, it's referenced in the PR's description (e.g. `Fix #xxx[,#xxx]`, where "xxx" is the Github issue number)
- [ ] All tests are passing
- [ ] New/updated tests are included
- [x] Changelog was modified

If adding a **new feature**, the PR's description includes:
- [x] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**
- Enh #5560

### CODE REVIEW

**Review by luke- — CHANGES_REQUESTED** [review]
Thank you for your contribution!

File: protected/humhub/modules/user/models/fieldtype/BaseType.php [hunk 1]
```diff
@@ -29,6 +29,8 @@
 class BaseType extends Model
 {
 
+    const EVENT_INIT = "fieldTypesInit";
```
[the remark this answers is not in the document]
Comment (KarlKeu00) [reply protected/humhub/modules/user/models/fieldtype/BaseType.php:47]: Done. I have also included an example.

File: protected/humhub/modules/user/models/fieldtype/BaseType.php [hunk 2]
```diff
@@ -59,6 +61,10 @@ class BaseType extends Model
      */
     public $canBeDirectoryFilter = false;
 
+    public function __construct() {
```
[the remark this answers is not in the document]
Comment (KarlKeu00) [reply protected/humhub/modules/user/models/fieldtype/BaseType.php]: Unfortunately, I have not yet understood the init method correctly, but I hope that it is now correct.

**Review by luke- — APPROVED** [review]

**Comment by luke-** [comment]
Should be merged: Can you please check?
https://github.com/humhub/humhub/commit/bfb6bfe54934e8f19b49dffb13ff2c47c2339f0f

**Pull request closed without merging**